Source/Core/Duality/Utility/Coroutines/CoroutineManager.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Duality.Utility.Coroutines
+{
+	/// <summary>
+	/// The Coroutine Manager
+	/// </summary>
+	public class CoroutineManager
+	{
+		private readonly Queue<Coroutine> pool = new Queue<Coroutine>(64);
+		
+		private Queue<Coroutine> currentCycle = new Queue<Coroutine>(256);
+		private Queue<Coroutine> nextCycle = new Queue<Coroutine>(256);
+
+		private readonly Dictionary<Coroutine, Exception> lastFrameErrors = new Dictionary<Coroutine, Exception>();
+
+		/// <summary>
+		/// Returns an IEnumerable of all currently active and scheduled Coroutines
+		/// </summary>
+		public IEnumerable<Coroutine> Coroutines
+		{
+			get { return this.currentCycle.Concat(this.nextCycle); }
+		}
+
+		/// <summary>
+		/// Returns an IEnumerable of the Exception encountered in the last cycle, with the relative Coroutine
+		/// </summary>
+		public IReadOnlyDictionary<Coroutine, Exception> LastFrameErrors
+		{
+			get { return this.lastFrameErrors; }
+		}
```

Comment:
I don't think this is necessary, given that each Coroutine already has its `LastException` property. Maybe remove it for now? If someone really needs that, they can easily emulate it by iterating `Coroutines` and checking each `LastException`.

Reply:
Your idea assumes that the user is keeping track of all Coroutines in variables, when (for how I used them until now), they are more of a fire-and-forget thing; you only keep them if you need to be able to cancel them for some reason.
Plus, the dictionary gives you a quick access to only those in error.

Reply:
Its also better to itterate over the `Coroutines` and check the `LastException` property because its lazy and doesn't have to happen on every frame. No need to always track it I think.

Reply:
My reasoning is: If the user cares about the state of the coroutine, including errors, they will keep track of the reference to query its state. If they _don't_ do that, they probably don't care about it (fire and forget) like you said - but if they don't care about their state, they probably don't care about any exceptions either, and they can still debug them via the automatically logged exception in `coroutine.Update()`.

Reply:
Fair enough, away it goes